nautobot/dcim/models/device_components.py

```diff
@@ -773,6 +779,87 @@ def parent(self):
         return self.device
 
 
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "statuses",
+    "webhooks",
+)
+class InterfaceRedundancyGroup(PrimaryModel, ConfigContextModel, StatusModel):  # pylint: disable=too-many-ancestors
+    """
+    A collection of Interfaces that supply a redundancy group for protocols like HSRP/VRRP.
+    """
+
+    name = models.CharField(max_length=100, unique=True)
+    slug = AutoSlugField(populate_from="name")
+    description = models.CharField(max_length=200, blank=True)
+    virtual_ip = models.ForeignKey(
+        to="ipam.IPAddress",
+        on_delete=models.CASCADE,
+        null=True,
+        related_name="interface_redundancy_virtual_ip",
+    )
+    interfaces = models.ManyToManyField(
+        to="dcim.Interface",
+        through="dcim.InterfaceRedundancyGroupAssociation",
+        related_name="redundancy_groups",
+        blank=True,
+    )
+    protocol = models.CharField(
+        max_length=50,
+        blank=True,
+        choices=InterfaceRedundancyGroupProtocolChoices,
+        verbose_name="Redundancy Protocol",
+    )
+    secrets_group = models.ForeignKey(
+        to="extras.SecretsGroup",
+        on_delete=models.SET_NULL,
+        default=None,
+        blank=True,
+        null=True,
+    )
+    group_id = models.CharField(max_length=50, blank=True)
+
+    class Meta:
+        """Meta class."""
+
+        ordering = ["name"]
+
+    def get_absolute_url(self):
+        """Return detail view for InterfaceRedundancyGroup."""
+        return reverse("dcim:interfaceredundancygroup", args=[self.id])
+
+    def __str__(self):
+        """Return a string representation of the instance."""
+        return self.name
```

Comment:
Let's not forget some of the items here: https://github.com/nautobot/nautobot/issues/2825#issuecomment-1591789168

Reply:
I have implemented all of the changes from the comment:
- Use StatusField instead of StatusModel (see: Make status fields non-nullable and fixup any existing invalid nulls #3785)
- Overload created to pre-emptively model 2.0 behavior (see: Change ChangeLoggedModel.created from a DateField to a DateTimeField #2076)
- Change group and groups fields to interface_redundancy_group(s) to be more explicit
- Drop slug (see Remove the usage of slug fields #2662); URLs use PK, and name is already unique for 2.0 use with natural keys
- Remove subscribers
- Remove ConfigContextModel, should only need to be a PrimaryModel